core/domain/state_domain.py

```diff
@@ -540,6 +587,53 @@ def from_dict(cls, interaction_dict):
             [Hint.from_dict(h) for h in interaction_dict['hints']],
             solution_dict)
 
+    def to_proto(self):
+        """Returns a proto representation of the interaction object.
+
+        Returns:
+            InteractionInstance|dict. The InteractionInstance proto object
+            if the id of the interaction matches else return an empty dict.
+        """
+        interaction = None
+        interaction_proto = {}
+        if self.id is not None:
+            interaction = (
```

Comment:
May want to make sure this doesn't get confused with the interaction instance (just based on the name).

Reply:
fixed